lgc/builder/ImageBuilder.cpp

```diff
@@ -465,7 +465,15 @@ Value *ImageBuilder::CreateImageLoad(Type *resultTy, unsigned dim, unsigned flag
     imageDescArgIndex = args.size();
     args.push_back(imageDesc);
     args.push_back(getInt32(tfe));
-    args.push_back(getInt32(((flags & ImageFlagCoherent) ? 1 : 0) | ((flags & ImageFlagVolatile) ? 2 : 0)));
+
+    // glc/slc/dlc bits
+    unsigned cachePolicyMask = (flags & ImageFlagVolatile) ? 2 : 0; // slc
```

Comment:
I don't think mapping `volatile` to SLC is correct. SLC mostly affects the allocation policy of L2, basically LRU vs. streaming -- LLVM by default sets SLC for non-temporal ops. That's not the same as volatile.

I don't think there's a reason for volatile to behave any different from coherent.

Reply:
Thanks, I will have a look at "volatile". Note: my patch does not change the behaviour w.r.t. volatile, I've only added "dlc" bit for "coherent".